flink-core/src/main/java/org/apache/flink/configuration/description/HtmlFormatter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration.description;
+
+/**
+ * Formatter that transforms {@link Description} into Html representation.
+ */
+public class HtmlFormatter extends Formatter {
+
+	@Override
+	protected void formatLink(StringBuilder state, String link, String description) {
+		state.append(String.format("<a href=\"%s\">%s</a>", link, description));
+	}
+
+	@Override
+	protected void formatLineBreak(StringBuilder state) {
+		state.append("<br/>");
+	}
+
+	@Override
+	protected void formatText(StringBuilder state, String format, String[] elements) {
+		String escapedFormat = escapeCharacters(format);
+		state.append(String.format(escapedFormat, elements));
+	}
+
+	@Override
+	protected void formatList(StringBuilder state, String[] entries) {
+		state.append("<ul>");
+		for (String entry : entries) {
+			state.append(String.format("<li>%s</li>", entry));
+		}
+		state.append("</ul>");
+	}
+
+	@Override
+	protected Formatter newInstance() {
+		return new HtmlFormatter();
+	}
+
+	private static final String TEMPORARY_PLACEHOLDER = "superRandomTemporaryPlaceholder";
+
+	private static String escapeCharacters(String value) {
+		return value
+			.replaceAll("%s", TEMPORARY_PLACEHOLDER)
```

Comment:
@zentol Could you have a last look on this block code, before I merge it? Didn't want to sneak it in. I found out there is an issue when we use '%' sign in the description.

Reply:
can you give me an example for where this is problematic? Does this occur if the _final formatted_ description contains `%`?

Reply:
Sure, it is problematic whenever we use the '%' symbol for percents e.g. 

	Description description = Description.builder()
		.text("This is a text that has some percentage value of 20%.")
		.build();

There are already some descriptions that break it e.g

>The relative amount of memory (after subtracting the amount of memory used by network buffers) that the task manager reserves for sorting, hash tables, and caching of intermediate results. For example, a value of `0.8` means that a task manager reserves 80% of its memory for internal data buffers, leaving 20% of free memory for the task manager's heap for objects created by user-defined functions. This parameter is only evaluated, if taskmanager.memory.size is not set.

Reply:
and what is the actual issue?

Reply:
That the `% o` is considered to be placeholder by `String.format(` therefore in this particular case `MissingFormatArgumentException` is thrown.

Reply:
ok that makes sense, do we maybe want to handle this in `Formatter´ already?.

Reply:
Actually I think it would make even more sense if I move it to `TextElement`

Reply:
Forget my comment, it should be done around formatting.